### ISSUE TITLE
Square on large complex(64) inputs return incorrect results

### ISSUE THREAD

**Issue by pearu** [jax-ml/jax#19751]
### Description

As reported in [MPMath vs Jax square, CPU](https://github.com/pearu/complex_function_validation/blob/main/mpmath_jax_results/data/square_MPMath_complex128_cpu_versus_JAX_complex64_cpu.txt) and [MPMath vs JAX square, CUDA](https://github.com/pearu/complex_function_validation/blob/main/mpmath_jax_results/data/square_MPMath_complex128_cpu_versus_JAX_complex64_cuda.txt), there exists regions in complex plane where `jax.numpy.square` returns incorrect results. For example:
```python
>>> with jax.default_device(jax.devices('cpu')[0]):
...   jnp.square(jnp.array(4.468666049354532e+32-5.8683295066770075e+26j, dtype=jnp.complex64))
... 
Array(-inf-infj, dtype=complex64)
>>> with jax.default_device(jax.devices('cuda')[0]):
...   jnp.square(jnp.array(4.468666049354532e+32-5.8683295066770075e+26j, dtype=jnp.complex64))
... 
Array(nan-infj, dtype=complex64)
```
The expected result is `inf - infj` as suggested by:
```python
>>> (4.468666049354532e+32-5.8683295066770075e+26j)**2
(1.9968976260619403e+65-5.244720966582595e+59j)
```
or
```python
>>> with jax.default_device(jax.devices('cpu')[0]):
...   jnp.square(jnp.array(4.893543125372159e+30-4.893543125372159e+30j, dtype=jnp.complex64))
... 
Array(-inf-infj, dtype=complex64)
```
where the correct result is `-infj`.

Other samples problematic to jax square include:
```
                                                      x ->      jax x**2   mpmath x**2
        (4.468666049354532e+32-5.8683295066770075e+26j) ->   (-inf-infj)   (inf-infj)
        (4.468666049354532e+32+5.8683295066770075e+26j) ->   (-inf+infj)   (inf+infj)
       (-4.468666049354532e+32+5.8683295066770075e+26j) ->   (-inf-infj)   (inf-infj)
                           (inf-5.358817286521974e+28j) ->    (nan-infj)   (inf-infj)
                           (inf+4.893543125372159e+30j) ->    (nan+infj)   (inf+infj)
                          (-inf-5.358817286521974e+28j) ->    (nan+infj)   (inf+infj)
                          (-inf+4.893543125372159e+30j) ->    (nan-infj)   (inf-infj)
         (4.893543125372159e+30-4.893543125372159e+30j) ->   (nan-infj)         -infj
         (4.893543125372159e+30+4.893543125372159e+30j) ->   (nan+infj)          infj
        (-4.893543125372159e+30-4.893543125372159e+30j) ->   (nan+infj)          infj
        (-4.893543125372159e+30+4.893543125372159e+30j) ->   (nan-infj)         -infj
```

### What jax/jaxlib version are you using?

0.4.23

### Which accelerator(s) are you using?

CPU/GPU

**Comment by pearu** [comment]
This issue is fixed via https://github.com/jax-ml/jax/pull/24874 . The current state of JAX `square` accuracy is summarized in https://github.com/pearu/functional_algorithms/issues/52.
